apps/wallet-mobile/src/NftDetails/NftDetailsImage.tsx

```diff
@@ -21,20 +20,25 @@ const isParams = (params?: Params | object | undefined): params is Params => {
 export const NftDetailsImage = () => {
   const {id} = useParams<Params>(isParams)
   const wallet = useSelectedWallet()
-  const nft = useNft(wallet, {id})
-  const {track} = useMetrics()
+  const dimensions = useWindowDimensions()
+
+  // reading from the getter, there is no need to subscribe to changes
+  const [amount] = React.useState(wallet.balances.records.get(id))
```

Comment:
yap, the balances is not an observable os if eventually the token is gone on next render it will be blank, there is no need to subscribe once is opened the details or the image, but further renders must not affect what is already loaded, so use state here is basically a memo.